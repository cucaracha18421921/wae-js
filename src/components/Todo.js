import React from 'react'
import PropTypes from 'prop-types'
import { splitTextIntoWords } from "../utils/textUtils";
import styled from "styled-components";

const Text = styled.a`
  font-size: xx-large;
`;

const HighlightedText = styled(Text)`
  background-color: yellow;
`;

const renderText = (text, highlightText ) => {
  if(!highlightText){
    return <Text>{text}</Text>;
  }
  const highlightedTextArray = splitTextIntoWords(text, highlightText);
  return highlightedTextArray.map(({text, highlight}) =>(highlight)?<HighlightedText>{text}</HighlightedText>:<Text>{text}</Text>)
};

const Todo = ({ onClick, completed, text, highlightText }) => (
  <li
    onClick={onClick}
    style={{
      textDecoration: completed ? 'line-through' : 'none'
    }}
  >
    {renderText(text,highlightText)}
  </li>
)

Todo.propTypes = {
  onClick: PropTypes.func.isRequired,
  completed: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired
}

export default Todo
