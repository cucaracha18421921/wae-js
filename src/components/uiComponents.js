import styled from "styled-components";

export const Button = styled.button`
    font-size: 0.875rem;
    min-width: 64px;
    box-sizing: border-box;
    transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
    font-family: "Roboto", "Helvetica", "Arial", sans-serif;
    font-weight: 500;
    line-height: 1.75;
    border-radius: 4px;
    letter-spacing: 0.02857em;
    text-transform: uppercase;
    color: #1976d2;
    border: 1px solid rgba(25, 118, 210, 0.5);
    background-color: transparent;
`;

export const Input = styled.input`
  line-height: 1.75;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  border-radius: 4px;
  border: 1px solid rgba(25, 118, 210, 0.5);
  &:focus {
    border: 1px solid rgba(25, 118, 210, 0.5);
    box-shadow: 0 0 0 1px rgba(25, 118, 210, 0.5);
    outline: none;
    }
`;


export const Text = styled.a`
  font-size: xx-large;
`;

export const HighlightedText = styled(Text)`
  background-color: yellow;
`;
