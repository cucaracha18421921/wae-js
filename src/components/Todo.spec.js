import {mount, shallow} from "enzyme";
import React from "react";
import Todo from "./Todo";
import {HighlightedText} from "./uiComponents";

describe('Rendering todo items', () => {
  const mockClickFunction = jest.fn();
  const wrapper = shallow(<Todo key={'todo'} onClick={mockClickFunction} completed={true} text={'search text'} highlightText={'search'}/>);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have textDecoration linethrough when task is completed',()=>{
    expect(wrapper.get(0).props.style.textDecoration).toEqual('line-through');
  });

  it('should have no linethrough when task is active',()=>{
    const wrapper = shallow(<Todo key={'todo'} onClick={mockClickFunction} completed={false} text={'search text'} highlightText={'search'}/>);
    expect(wrapper.get(0).props.style.textDecoration).toEqual('none');
  });

  it('should highlight text when is a searchKey',()=>{
    const wrapper = mount(<Todo key={'todo'} onClick={mockClickFunction} completed={false} text={'search'} highlightText={'search'}/>);
    const yellowTextWrapper = mount(<HighlightedText>{'search'}</HighlightedText>);
    expect(yellowTextWrapper.find('a').props().className).toEqual(wrapper.find('a').props().className);
  })
});
