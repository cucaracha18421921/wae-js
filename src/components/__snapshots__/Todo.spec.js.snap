// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link rendering renders correctly 1`] = `ShallowWrapper {}`;

exports[`Rendering todo items renders correctly 1`] = `ShallowWrapper {}`;
